ManageRDSessions/ManageRDSessions.psm1

```diff
@@ -1,71 +1,108 @@
-function Get-RDActiveSession {
-    $sessions = Get-RDUserSession | Where-Object { $_.SessionState -eq 'STATE_ACTIVE' -or $_.SessionState -eq 'STATE_IDLE' }
-    $sessions | Select-Object HostServer, UserName, UnifiedSessionId, SessionState | Format-Table
-}
+function Get-sbRDSession {
+    [OutputType('Custom.SB.RDSession')]
+    [cmdletbinding()]
+    param(
+        [Parameter(Mandatory = $false)]
+        [ValidateSet("Active", "Idle", "Connected", "Disconnected", "Any")]
+        [Alias("State")]
+        [string]$SessionState = "Any",
 
-function Get-RDDisconnectedSession {
-    [CmdletBinding()]
-    param (
-        [
-        Parameter(
-            Mandatory = $False,
-            HelpMessage = "Specify minimum number of idle session minutes"
-        )
-        ]
-        [Alias('idle')]
-        [int]$MinIdleTime = 0
+        [Parameter(Mandatory = $false)]
+        [int]$MinimumIdleTime = 0
     )
 
-    $sessions = Get-RDUserSession | Where-Object { $_.SessionState -eq 'STATE_DISCONNECTED' -and ( $_.IdleTime / 60000 ) -gt $MinIdleTime }
-    $sessions | Select-Object HostServer, UserName, @{ Label = 'Idle Time (Mins)'; expression = {$_.IdleTime / 60000 -as [int] } }, UnifiedSessionId, SessionState | Format-Table
-}
+    Begin {
+        # Intentionally empty
+    }
+    # Process block is required when outputting objects to the pipeline, otherwise only the final object will be passed
+    Process {
 
-function Remove-RDDisconnectedSession {
-    [CmdletBinding()]
-    param (
-        [
-        Parameter(
-            Mandatory = $False,
-            HelpMessage = "Specify minimum number of idle session minutes"
-        )
-        ]
-        [Alias('idle')]
-        [int]$MinIdleTime = 0,
-        [switch]$Force = $False
-    )
+        $state = switch ($SessionState) {
```

Comment:
Good use of switch.
However, if you're not doing extra in each lines scriptblock, you could also use a hashtable like this:
```
$stateLookup = @{
    Active       = "STATE_ACTIVE" 
    Idle         = "STATE_IDLE" 
    Disconnected = "STATE_DISCONNECTED" 
    Connected    = "STATE_CONNECTED" 
    Any          = "*" 
}
```
Then use like this:
```
$sessions = Get-RDUserSession | Where-Object {
    $_.SessionState -like $stateLookup.$SessionState -and ( $_.IdleTime / 60000 ) -ge $MinimumIdleTime
}
```

Reply:
LOVE this. Thank you.